src/braket/pulse/waveforms.py

```diff
@@ -85,25 +86,31 @@ def __init__(self, amplitudes: list[complex], id: Optional[str] = None):
         self.amplitudes = list(amplitudes)
         self.id = id or _make_identifier_name()
 
-    def __eq__(self, other):
+    def __eq__(self, other: ArbitraryWaveform):
```

Comment:
If we type check here, do we need the isInstance call later?

Reply:
This is [type hinting](https://docs.python.org/3/library/typing.html). While we specific that `other` should be an `ArbitraryWaveform`, you can still pass in arbitrary objects.